src/sentry/web/frontend/accounts.py

```diff
@@ -123,7 +123,13 @@ def recover_confirm(request, user_id, hash):
 @login_required
 def start_confirm_email(request):
     has_unverified_emails = request.user.has_unverified_emails()
-    if has_unverified_emails:
+
+    if 'email' in request.GET:
```

Comment:
I went ahead and created a separate method for sending the confirmation for primary email since the primary email value doesn't change and therefore doesn't need to be passed as a parameter. With this change, would this still run the risk of causing emails to be generated?